src/server/scripts/Northrend/zone_grizzly_hills.cpp

```diff
@@ -934,6 +935,254 @@ class spell_infected_worgen_bite : public SpellScriptLoader
     }
 };
 
+/*######
+## Quest: Riding the Red Rocket
+######*/
+
+enum RedRocket
+{
+    SPELL_VEHICLE_WARHEAD_FUSE = 49107,
+    SPELL_ALLIANCE_KILL_CREDIT_TORPEDO = 49510,
+    SPELL_HORDE_KILL_CREDIT_TORPEDO = 49340,
+    NPC_HORDE_LUMBERBOAT = 27702,
+    NPC_ALLIANCE_LUMBERBOAT = 27688,
+    SPELL_DETONATE = 49250
+};
+
+class npc_rocket_propelled_warhead : public CreatureScript
+{
+public:
+    npc_rocket_propelled_warhead() : CreatureScript("npc_rocket_propelled_warhead") { }
+
+    struct npc_rocket_propelled_warheadAI : public VehicleAI
+    {
+        npc_rocket_propelled_warheadAI(Creature* creature) : VehicleAI(creature)
+        {
+            _finished = false;
+            _faction = ALLIANCE;
+        }
+
+        void PassengerBoarded(Unit* who, int8 /*seatId*/, bool apply) override
+        {
+            if (apply && who->ToPlayer())
```

Comment:
shouldn't we check `who` too? `apply && who && who->ToPlayer()`

Reply:
It makes sense to check who just in case but it should never be null as a unit would always board it in theory 🗡️ 
Will add the check to be safe and aswell ass all the missing brackets